chain/consensus/hierarchical/subnet/resolver/resolver.go

```diff
@@ -346,24 +362,104 @@ func (r *Resolver) processResponse(ctx context.Context, rmsg *ResolveMsg) (pubsu
 	return pubsub.ValidationAccept, nil
 }
 
-func (r *Resolver) getLocal(ctx context.Context, c cid.Cid) (*sca.CrossMsgs, bool, error) {
+func (r *Resolver) processPullLocked(submgr subnet.SubnetMgr, rmsg *ResolveMsg) (pubsub.ValidationResult, error) {
+	// FIXME: Make this configurable
+	lstate, found, err := r.getLockedStateFromActor(context.TODO(), submgr, rmsg)
+	if err != nil {
+		return pubsub.ValidationIgnore, xerrors.Errorf("error geting locked state from actor: %s", err)
+	}
+	if !found {
+		// Reject instead of ignore. Someone may be trying to spam us with
+		// random unvalid CIDs.
+		return pubsub.ValidationReject, xerrors.Errorf("couldn't find lockedState with cid: %s", rmsg.Cid)
+	}
+	// Send response
+	if err := r.publishLockedResponse(*lstate, rmsg.From); err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	// Publish a Response message to the source subnet if the CID is found.
+	return pubsub.ValidationAccept, nil
+}
+
+func (r *Resolver) publishLockedResponse(lstate atomic.LockedState, to address.SubnetID) error {
+	c, err := lstate.Cid()
+	if err != nil {
+		return err
+	}
+	m := &ResolveMsg{
+		Type:   ResponseLocked,
+		From:   r.netName,
+		Cid:    c,
+		Locked: lstate,
+	}
+	return r.publishMsg(m, to)
```

Comment:
I would recommend
```
m := ResolveMsg{
	Type:   ResponseLocked,
	From:   r.netName,
	Cid:    c,
	Locked: lstate,
}
return r.publishMsg(&m, to)
```

Reply:
Fixed, but out of a curiosity, is this for performance or aesthetic reasons?

Reply:
This is to improve code readability.
The citation from Bill Kennedy's Go Playbook:
```
When constructing a variable, never use pointer semantics.
In other words, don’t construct to a pointer.
Construction tells you nothing about what the escape analysis algorithms will do.
It does not matter if the API you will be using wants a pointer to these values. Keep the data semantics clean.
The ampersand operator is an important readability marker for how a value is being shared.
```

This line of code tells you the value has the potential to be constructed on the heap since it is being shared up the call stack: `return &m`

This return tells you nothing without reading more code to understand what the `m` variable is: 
`return m`

Does it make sense?